components/esp_modem/src/esp_modem_command_library.cpp

```diff
@@ -10,15 +10,17 @@
 #include "cxx_include/esp_modem_dte.hpp"
 #include "cxx_include/esp_modem_dce_module.hpp"
 #include "cxx_include/esp_modem_command_library.hpp"
+#include "cxx_include/esp_modem_command_library_utils.hpp"
+
 
 namespace esp_modem::dce_commands {
 
 static const char *TAG = "command_lib";
 
-command_result generic_command(CommandableIf *t, const std::string &command,
-                               const std::list<std::string_view> &pass_phrase,
-                               const std::list<std::string_view> &fail_phrase,
-                               uint32_t timeout_ms)
+static command_result generic_command(CommandableIf *t, const std::string &command,
```

Comment:
Nitpick: `static` function could be in an unnamed namespace.

Reply:
Tried to enclose it in an anonymous namespace, but IMHO it makes the module less readable, as I'd have to split the API to `static`s and "others". I would keep it as is, if you agree, since they're already enclosed in their own namespaces.